scripts/upload-source-maps.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+import { execSync } from 'child_process'
+
+const main = () => {
+	if (process.env.VERCEL_ENV === 'development') {
+		return
+	}
+
+	const LATEST_SHA = process.env.VERCEL_GIT_COMMIT_SHA
+	const PATH_PREFIX = 'https://developer.hashicorp.com/_next/static/'
+	const SERVICE = 'developer.hashicorp.com'
+
+	const DATADOG_API_KEY = process.env.DD_API_KEY
+
+	try {
+		const status = execSync(
+			`DATADOG_API_KEY=${DATADOG_API_KEY} npx @datadog/datadog-ci sourcemaps upload .next/static/ --service=${SERVICE} --release-version=${LATEST_SHA} --minified-path-prefix=${PATH_PREFIX}`
```

Comment:
Noticed this error in the Vercel build logs:

![CleanShot 2024-07-25 at 11 11 34@2x](https://github.com/user-attachments/assets/6f75a527-665a-43f8-89a2-103b86d9e03d)

It seems like "running `datadog-ci` within the git repo" should in theory be happening, but isn't working as expected 🤔 It seems like we can [manually set the relevant info ourselves](https://github.com/DataDog/datadog-ci/tree/master/src/commands/sourcemaps#link-errors-with-your-source-code)... but I am curious why the DataDog CLI doesn't seem to be finding any git remotes...  maybe a Vercel thing? 🤷 I'm lost 😬

Reply:
I think this is coming up it's trying to get metadata. From the DD docs.

> If you run datadog-ci sourcemaps upload within a Git working directory, Datadog collects repository metadata. 
More info here: https://github.com/DataDog/datadog-ci/tree/master/src/commands/sourcemaps#link-errors-with-your-source-code

Reply:
Update: the error went away when I added the flag `--disable-git` on my test branch

Reply:
Sweet, glad the error's gone! 🎉

In a perfect world, I think if we could resolve the issue without the `--disable-git` flag, then we might be able to link back to the source repo (screenshot from [here](https://app.datadoghq.com/rum/sessions?query=%40type%3Aerror%20%40application.id%3A40a2019f-88ad-4076-ab52-ffd074064eed%20-%40error.source%3Anetwork%20%40session.type%3Auser%20version%3A3e2af5b85963f3a28d7570ae060990aa512f0695&agg_m=count&agg_m_source=base&agg_t=count&cols=&event=AgAAAZDrLNNJeFyzhAAAAAAAAAAYAAAAAEFaRHJMTmM2QUFBMTd0aDVtUTZmM2dBTQAAACQAAAAAMDE5MGViMmMtZDUwYy00OTc1LThiMzYtYTJlYzkzNjA1MTI2&fromUser=false&refresh_mode=sliding&track=rum&viz=stream&from_ts=1721327766712&to_ts=1721932566712&live=true)):

![CleanShot 2024-07-26 at 12 02 52@2x](https://github.com/user-attachments/assets/8ff08bac-b24b-4259-a566-8237da7aaba0)

Feels kinda unnecessary though... the setup you've done to get the above link working feels like it's the really valuable thing here 👍 Finding out how to get DataDog's `git` stuff to work feels like something we can tack on later as a very optional enhancement if we really want to 👍